src/JustSaying/Fluent/PublishConfig/StaticPublicationConfiguration.cs

```diff
@@ -0,0 +1,87 @@
+using Amazon;
+using Amazon.Internal;
+using Amazon.SimpleNotificationService.Model;
+using JustSaying.AwsTools;
+using JustSaying.AwsTools.MessageHandling;
+using JustSaying.AwsTools.QueueCreation;
+using JustSaying.Messaging;
+using Microsoft.Extensions.Logging;
+
+#pragma warning disable CS0618
+
+namespace JustSaying.Fluent;
+
+internal class StaticPublicationConfiguration : TopicPublisher
+{
+    public Func<CancellationToken, Task> StartupTask { get; }
+    public IMessagePublisher Publisher { get; }
+
+    public StaticPublicationConfiguration(
+        Func<CancellationToken, Task> startupTask,
+        IMessagePublisher publisher)
+    {
+        StartupTask = startupTask;
+        Publisher = publisher;
+    }
+
+    public static StaticPublicationConfiguration Build<T>(
+        string topicName,
+        Dictionary<string, string> tags,
+        string region,
+        SnsWriteConfiguration writeConfiguration,
+        IAwsClientFactoryProxy proxy,
+        ILoggerFactory loggerFactory,
+        JustSayingBus bus)
+    {
+        var readConfiguration = new SqsReadConfiguration(SubscriptionType.ToTopic)
+        {
+            TopicName = topicName
+        };
+
+        readConfiguration.ApplyTopicNamingConvention<T>(bus.Config.TopicNamingConvention);
+
+        var eventPublisher = new SnsMessagePublisher(
+            proxy.GetAwsClientFactory().GetSnsClient(RegionEndpoint.GetBySystemName(region)),
+            bus.SerializationRegister,
+            loggerFactory,
+            bus.Config.MessageSubjectProvider)
+        {
+            MessageResponseLogger = bus.Config.MessageResponseLogger,
+        };
+
+        var snsTopic = new SnsTopicByName(
+            readConfiguration.TopicName,
+            proxy.GetAwsClientFactory().GetSnsClient(RegionEndpoint.GetBySystemName(region)),
```

Comment:
Should they share the client, rather than have one each?

Reply:
It's an interesting one! I don't see why not - seems like a waste. We do currently have a different client per publisher for other topics, but we should probably be sharing them more. Will do this 👍

Reply:
Done 👍